### ISSUE TITLE
DASH manifest wrongly generated

### ISSUE THREAD

**Issue by simtrami** [OpenVisualCloud/CDN-Transcode-Sample#140]
**Bug description**
The manifest `index.mpd` generated by the transcoding process does not allow automatic switching in resolutions because it is badly formed.  
This issue is written after making sure #10 and #19 and #26 didn't cover it.

**To Reproduce**
1. Set up the vod transcode service ([example](https://github.com/Ressource-Allocation/CDN-Transcode-Sample/wiki/Deployment-configuration))
2. Start a DASH stream to begin the transcoding process (from the web GUI, for instance)
3. On the deployment server (host), inspect the generated manifest. From the repository folder : `less volume/dash/<stream_name>/index.mpd`

**Expected behavior**
The manifest should contain one single `<AdaptationSet>...</AdaptationSet>` tag with the attribute `contentType="video"`, containing all the `<Representation>...</Representation>` tags, each corresponding to a resolution. **This is not the case** as it has one `<AdaptationSet>...</AdaptationSet>` per resolution, each containing a single `<Representation>...</Representation>`. This results in not being able to switch automatically when ABR logic would need it.

You can test in depth the ABR behaviour with [this standard DASH player](https://reference.dashif.org/dash.js/nightly/samples/dash-if-reference-player/index.html). Give it a URL formed this way: `https://<host IP>/dash/<stream_name>/index.mpd`.

**Screenshots**
[DASH manifest screenshot](https://pasteboard.co/J250cko.png)

**Server:**
 - OS: Ubuntu Server 18.04
 - Configuration process: Described [here](https://github.com/Ressource-Allocation/CDN-Transcode-Sample/wiki/Deployment)
 - Version: 63e4b8577371050791aaee7ff084235ca62ae912

**Additional context**
We forked your project for a small college research on edge-computing ([more info](https://github.com/Ressource-Allocation/CDN-Transcode-Sample/wiki)). Solving this issue is a matter of efficiency as we need to correct the manifest by hand every time.

**Comment by QuPengfei** [comment]
@simtrami  Please try the patch.

[0001-one-adaptation_sets-for-all-the-resolution-in-dash-m.zip](https://github.com/OpenVisualCloud/CDN-Transcode-Sample/files/4750864/0001-one-adaptation_sets-for-all-the-resolution-in-dash-m.zip)

**Comment by simtrami** [comment]
Thank you!

I tried pulling this fix one my own but it kept failing when adding the `-adaptation_sets` flag in the `call()` arguments list.  
Your solution makes me figure out I vainly tried putting quotes around `id=0,streams=v` - as it should be if it was executed from terminal (eg. `-adaptation_sets "id=0,streams=v"`) - however, with the `subprocess.call` function, it behaves as if I wrote `"'id=0,streams=v'"` and thus fails to parse, returning a non-helpful message.

It is quite a simple fix and the ffmpeg doc makes it easily understandable ([here](https://ffmpeg.org/ffmpeg-formats.html#dash-2)) but I lost so much time on this abnormally long lasting issue.

I'll pull [your changes](https://github.com/QuPengfei/CDN-Transcode-Sample/commit/9a31a8f3c65a44e851340d645a3aac4d9afd5641) to our solution.
